### PR TITLE
Display block type in reviewer tools & Use specific activity log for soft-blocking versions

### PR DESCRIPTION
**Pull request by diox**
Fixes: mozilla/addons#15125

### Testing

Add some hard and soft-blocks / harden some soft-blocks / soften some hard-blocks: this should be reflected in reviewer tools review page next to the corresponding version in the history. (For accurate soft-block info, the block need to have been created as soft or softened after applying this patch)

### Screenshots

Version soft-blocked:
Screenshot 2024-11-25 at 13-55-45 rabbil-rabugong-narkato – Add-ons for Firefox

Version soft-blocked then hardened:
Screenshot 2024-11-25 at 14-23-01 rabbil-rabugong-narkato – Add-ons for Firefox

### CODE REVIEW

File: src/olympia/blocklist/utils.py [hunk 1]
```diff
@@ -38,23 +39,22 @@ def block_activity_log_save(
...
     if submission_obj:
         details['signoff_state'] = submission_obj.SIGNOFF_STATES.for_value(
             submission_obj.signoff_state
         ).display
         if submission_obj.signoff_by:
             details['signoff_by'] = submission_obj.signoff_by.id
-        details['soft'] = version_details['soft'] = (
-            submission_obj.block_type == BlockType.SOFT_BLOCKED
-        )
+        details['soft'] = submission_obj.block_type == BlockType.SOFT_BLOCKED
```
[the remark this answers is not in the document]
Comment (diox) [reply src/olympia/blocklist/utils.py:48]: Yeah I didn't love this either... There is no default value for `details['soft']` set earlier (there is for `action`) though, so I can't just set it there without a `else`...

I could pass `block_type`, it's only used in the admin. But then we would have something like:
```python
        details['block_type'] = submission_obj.block_type
        if submission_obj.block_type == BlockType.SOFT_BLOCKED:
            action_version = amo.LOG.BLOCKLIST_VERSION_SOFT_BLOCKED
```

Would that work for you ?

[the remark this answers is not in the document]
Comment (diox) [reply src/olympia/blocklist/utils.py]: I've pushed that, with necessary adjustments to the admin & tests: 39852d1c059fe1b10c30edb10aca3135dd8e6143 - let me know what you think

[the remark this answers is not in the document]
Comment (KevinMind) [reply src/olympia/blocklist/utils.py]: Looks good.

**Comment by KevinMind** [comment]
Soft -> Hard -> Soft

image

Hard -> Soft -> Hard -> Unblocked

image

**Review by KevinMind — APPROVED** [review]
Works.

**Pull request merged**